packages/core/src/browser/quick-open/quick-title-bar.ts

```diff
@@ -0,0 +1,249 @@
+/********************************************************************************
+ * Copyright (C) 2019 Red Hat, Inc. and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { Emitter, Event } from '../../common/event';
+import { injectable } from 'inversify';
+
+export enum QuickInputButtonSide {
+    LEFT = 0,
+    RIGHT = 1
+}
+
+export interface QuickTitleButton {
+    icon: string; // a background image coming from a url
+    iconClass?: string; // a class such as one coming from font awesome
+    tooltip?: string | undefined;
+    side: QuickInputButtonSide
+}
+
+@injectable()
+export class QuickTitleBar {
+
+    private readonly onDidTriggerButtonEmitter: Emitter<QuickTitleButton>;
+    private _isAttached: boolean;
+
+    private titleElement: HTMLElement;
+    private titleBarContainer: HTMLElement;
+    private attachedNode: HTMLElement | undefined;
+
+    private _title: string | undefined;
+    private _step: number | undefined;
+    private _totalSteps: number | undefined;
+    private _buttons: ReadonlyArray<QuickTitleButton>;
+
+    private tabIndex = 2; // Keep track of the tabIndex for the buttons
```

Comment:
`tabIndex` greater than 0 should not be used, it is common anti-pattern, since it breaks the natural focus flow.

Why `tabIndex=0` does not work to let browser decide how to give a focus according to the document flow?

Reply:
If it is hard to fix now, comment [here](https://github.com/theia-ide/theia/issues/5858) that the quick title bar should be reworked as well.

Reply:
`tabIndex=0` closes the `quick-pick` when `tab` is pressed, I'll comment the issue when the PR will be merged, to be able to keep references to master.